packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -854,9 +858,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         private _storage?: IDocumentStorageService,
     ) {
         super();
-
         this.baseSummaryMessage = metadata?.message;
 
+        /**
+         * If this is a new container, we initialize these values. If it's an existing container,
+         * these values would stay undefined
+         */
+        this.createContainerMetadata.createContainerRuntimeVersion = existing
```

Comment:
It's easier to read the code if it has two branches (in your case - there is need for two - the else branch is empty):

if (!existing) {
   this.createContainerMetadata = {
    createContainerRuntimeVersion : pkgVersion, ... };
}

Reply:
(should be ` = metadata ?? {}` I think, to maintain behavior)

Reply:
But she doesn't want all of `metadata`.  Although if she doesn't do the `...this.createContainerMetadata` suggestion in the log event it probably wouldn't matter :P

Reply:
But anyway, switching to `if` is still nice.

Reply:
Hello from over here in the weeds... Maybe JS needs a new function `Object.pick`:

```
function pick<T>(source: T, ...props: (keyof T)[]): Partial<T> {
    const initial: Partial<T> = {};
    return props.reduce((acc, p) => { acc[p] = source[p]; return acc; }, initial);
}
```

Then you could do `pick(metadata ?? {}, "createContainerRuntimeVersion", "createContainerTimestamp", "summaryCount")`

Ok back to reality...

You could do this:

```
const { createContainerRuntimeVersion, createContainerTimestamp, summaryCount } = existing
  ? metadata ?? {}
  : {
      createContainerRuntimeVersion: pkgVersion,
      createContainerTimestamp: performance.now(),
      summaryCount: 0,
   };
this.createContainerMetadata = { createContainerRuntimeVersion, createContainerTimestamp, summaryCount };
```

or do

```
if (existing) {
  this.createContainerMetadata.foo = metadata?.foo
  .. for each prop ..
} else {
  this.createContainerMetadata = { .. the values .. };
}
```

Or just don't change what you already have here 😆

Reply:
for me the third approach seems more readable, so i will do that one